src/map/skill.cpp

```diff
@@ -18465,6 +18465,18 @@ struct s_skill_condition skill_get_requirement(map_session_data* sd, uint16 skil
 						req.itemid[0] = skill->require.itemid[skill_lv - 1];
 						req.amount[0] = skill->require.amount[skill_lv - 1];
```

Comment:
I believe the issue is we write `req.itemid[0]` and `req.amount[0]` when `i = 1`
so we should only run this code if `i == 0`

Reply:
Hi @vstumpf,
I believe that changes is unrelated or not the fix.

The issue was it did not have a condition to make the `req.itemid[0] = req.amount[0] = 0` which result to no Gemstone will be consume by the skill.

Reply:
It does set it to 0; the issue is we go through the loop twice, and both times we set `itemid[0]` to the level requirement.

The first loop iteration (i = 0) sets req.itemid[0] correctly, then checks req.itemid[i] for gemstones.
The second iteration (i = 1) sets req.itemid[0] again, then checks req.itemid[i] for gemstones. 

So this is the reason why we should only set req.itemid[0] once on line 18464

Reply:
I see, then i will change code in line `18464` to below ?
`if (level_dependent && skill_lv <= MAX_SKILL_ITEM_REQUIRE && i == 0)`

Reply:
Yup! That's perfect

Reply:
Thank you @vstumpf 😄 ,
done added & commited.